### DISCUSSION TITLE
目前 Rive 還沒有官方專門為 Vue 框架提供的「Vue-specific runtime」

### DISCUSSION THREAD

**Discussion by leechiuhui** [3dw/freemath#81] [Q&A] [answered]
@bestian 老師 和 @Tofuswang 豆腐好： 

根據https://help.rive.app/runtimes/overview/vue 這裡的說明(如下圖)

<img width="1062" alt="截圖 2024-09-09 下午6 10 06" src="https://github.com/user-attachments/assets/8ffa4d3d-db1d-4179-9b45-015fa7392088">

，理解到目前 Rive 還沒有官方專門為 Vue 框架提供的「Vue-specific runtime」

但是經測試，我確認已經可以用下圖這段程式碼嵌入Rive的動畫。

<img width="1074" alt="截圖 2024-09-09 下午6 00 14" src="https://github.com/user-attachments/assets/f946c4f0-b96b-42a0-918f-d083ebb8cd1f">

但是我目前不太知道和不確定的是：
如果目前 Rive 還沒有官方專門為 Vue 框架提供的「Vue-specific runtime」的話，
不知道可不可以做到[[這篇文](https://yujih.substack.com/p/riveduolingo?utm_source=substack&publication_id=1078938&post_id=148464549&utm_medium=email&utm_content=share&utm_campaign=email-share&triggerShare=true&isFreemail=true&r=3imjca&triedRedirect=true)]提到的

**1、跨平台互動播放器：### 可以根據使用者的行為，播放對應的動畫**

<img width="994" alt="截圖 2024-09-09 下午6 02 58" src="https://github.com/user-attachments/assets/f22e7357-0959-462b-8887-417413cf60d5"> 

和

 **2、### 貓咪的手會根據游標的位置和停留時間，而有不同動作。**

<img width="783" alt="截圖 2024-09-09 下午6 06 09" src="https://github.com/user-attachments/assets/6a21e82b-bf21-4097-ab2b-ebd51a2877f7">

和

**3、### 「根據使用者的行為，決定該切換到哪一個動畫，即動畫師設計狀態，並根據條件（游標位置、長按、點擊等）觸發。**

<img width="788" alt="截圖 2024-09-09 下午6 08 17" src="https://github.com/user-attachments/assets/b8e9c8a9-3050-4021-bafa-480f597452a2">

我在想：如果接下來我們測試用Ｖue程式碼操作Rive達到「根據使用者的行為，播放對應的動畫的話」是可行的話，我可以負責去升級Rive的協作為Voyager方案（可無限的文件和專案）....

再看看Bestian老師和豆腐和大家的意見如何....感謝～：））

**Comment by bestian** [comment, marked as answer]
@leechiuhui 

個人的看法是，如果只是網頁上簡單的event-driven animation，像Rive首頁上的爪子，可以用疊圖、vue的:class和css的transition, animation來實現。
我想Rive的強效是可以做有結構較複雜或時間軸較長的animation.

因為我之前只學過Flash的影格和補間, 以及css的transition, animation，還不確定整合這兩者的Rive可以拿來做什麼題目。

建議先合作把這個小題目的後半部設計完成，這只需要影格和補間，不需要event-driven.
等新的題目出現，再考慮升級方案，可以節省一些開銷~

[草稿1](https://github.com/3dw/freemath/wiki/%E5%8B%95%E7%95%AB%E8%A8%AD%E8%A8%88%E8%8D%89%E7%A8%BF1:-%E4%BB%A5%E9%9D%A2%E7%A9%8D%E7%9A%84%E8%A7%92%E5%BA%A6%E7%9C%8B%E4%B9%98%E6%B3%95)

**Reply by bestian** [reply]
另外這個模板可以參考看看，好像是用vue的state managent來搭配Rive動畫，兩者我都不熟，只有模糊的概念
可能要做具體的題目，才能練出功力~

https://github.com/flowreaction/rive-vue
